service-templates/apigw-lambda-svc/v1/pipeline_infrastructure/pipeliine.tf

```diff
@@ -0,0 +1,685 @@
+        aws_iam_role.pipeline.arn
+      ]
+      type = "AWS"
+    }
+  }
+}
+
+################################################################################
+# IAM Role - Build
+################################################################################
+
+resource "aws_iam_role" "build" {
+  name_prefix = "build-role"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "codebuild.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+data "aws_iam_policy_document" "build" {
+  statement {
+    effect = "Allow"
+    resources = [
+      "arn:aws:logs:${local.region}:${local.account_id}:log-group:/aws/codebuild/${aws_codebuild_project.build.name}",
+      "arn:aws:logs:${local.region}:${local.account_id}:log-group:/aws/codebuild/${aws_codebuild_project.build.name}*"
+    ]
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+  }
+  statement {
+    effect = "Allow"
+    resources = [
+      "arn:aws:codebuild:${local.region}:${local.account_id}:report-group:/${aws_codebuild_project.build.name}*",
+    ]
+    actions = [
+      "codebuild:CreateReportGroup",
+      "codebuild:CreateReport",
+      "codebuild:UpdateReport",
+      "codebuild:BatchPutTestCases"
+    ]
+  }
+  statement {
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["proton:GetService"]
+  }
+  statement {
+    effect = "Allow"
+    resources = [
+      aws_s3_bucket.function.arn,
+      "${aws_s3_bucket.function.arn}/*"
+    ]
+    actions = [
+      "s3:GetObject*",
+      "s3:GetBucket*",
+      "s3:List*",
+      "s3:DeleteObject*",
+      "s3:PutObject*",
+      "s3:Abort*",
+      "s3:CreateMultipartUpload"
+    ]
+  }
+  statement {
+    effect = "Allow"
+    resources = [
+      aws_s3_bucket.pipeline_artifacts.arn,
+      "${aws_s3_bucket.pipeline_artifacts.arn}*"
+    ]
+    actions = [
+      "s3:GetObject*",
+      "s3:GetBucket*",
+      "s3:List*",
+      "s3:DeleteObject*",
+      "s3:PutObject*",
+      "s3:Abort*"
+    ]
+  }
+  statement {
+    effect    = "Allow"
+    resources = [aws_kms_key.pipeline_artifacts.arn]
+    actions = [
+      "kms:Decrypt",
+      "kms:DescribeKey",
+      "kms:Encrypt",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "build" {
+  policy = data.aws_iam_policy_document.build.json
+}
+
+resource "aws_iam_role_policy_attachment" "build" {
+  policy_arn = aws_iam_policy.build.arn
+  role       = aws_iam_role.build.name
+}
+
+################################################################################
+# CodeBuild Project - Build
+################################################################################
+
+locals {
+  # Lookup map that converts shorthand syntax to expanded form CodePipeline accepts
+  buildspec_runtime = {
```

Comment:
I assume this map+lookup is used due to the use of the lambda runtime and converting it to the codebuild equivalent?